Detectors/TPC/calibration/SpacePoints/include/SpacePoints/TrackInterpolation.h

```diff
@@ -0,0 +1,231 @@
+  std::array<float, Constants::MAXGLOBALPADROW> clAngle{};               ///< alpha for TPC cluster
+  std::array<unsigned short, Constants::MAXGLOBALPADROW> clAvailable{};  ///< if a TPC cluster is available in a given row this is set to 1
+  std::array<float, Constants::MAXGLOBALPADROW> phi{};                   ///< track phi (taken from extrapolation)
+  std::array<float, Constants::MAXGLOBALPADROW> tgl{};                   ///< track tgl (taken from extrapolation)
+};
+
+/// Structure used to store the TPC cluster residuals
+struct TPCClusterResiduals {
+  short dY{};           ///< residual in Y
+  short dZ{};           ///< residual in Z
+  short y{};            ///< Y position of track
+  short z{};            ///< Z position of track
+  short phi{};          ///< phi angle of track
+  short tgl{};          ///< dip angle of track
+  unsigned char sec{};  ///< sector number 0..17
+  unsigned char dRow{}; ///< distance to previous row in units of pad rows
+  short row{};          ///< TPC pad row (absolute units)
+  void setDY(float val) { dY = fabs(val) < param::sMaxResid ? static_cast<short>(val * 0x7fff / param::sMaxResid) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+  void setDZ(float val) { dZ = fabs(val) < param::sMaxResid ? static_cast<short>(val * 0x7fff / param::sMaxResid) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+  void setY(float val) { y = fabs(val) < param::sMaxY ? static_cast<short>(val * 0x7fff / param::sMaxY) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+  void setZ(float val) { z = fabs(val) < param::sMaxZ ? static_cast<short>(val * 0x7fff / param::sMaxZ) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+  void setPhi(float val) { phi = fabs(val) < param::sMaxAngle ? static_cast<short>(val * 0x7fff / param::sMaxAngle) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+  void setTgl(float val) { tgl = fabs(val) < param::sMaxAngle ? static_cast<short>(val * 0x7fff / param::sMaxAngle) : static_cast<short>(std::copysign(1., val) * 0x7fff); }
+};
+
+/// Structure filled for each track with track quality information and a vector with TPCClusterResiduals
+struct TrackData {
+  int trkId{};                              ///< track ID for debugging
+  float eta{};                              ///< track dip angle
+  float phi{};                              ///< track azimuthal angle
+  float qPt{};                              ///< track q/pT
+  float chi2TPC{};                          ///< chi2 of TPC track
+  float chi2ITS{};                          ///< chi2 of ITS track
+  unsigned short nClsTPC{};                 ///< number of attached TPC clusters
+  unsigned short nClsITS{};                 ///< number of attached ITS clusters
+  int nTracksInEvent{};                     ///< total number of tracks in given event / timeframe ? Used for multiplicity estimate
+  std::vector<TPCClusterResiduals> clRes{}; ///< residuals for each available TPC cluster of this track
```

Comment:
below you are using vector<TrackData>. Using large vectors of vectors will have overheads. I would split it to vector<TPCClusterResiduals> (stored in a separate tree) covering many tracks and separate vector<TrackData>, with TrackData (in a separate tree) holding the reference on the entry of its 1st TPCClusterResiduals and their number, see e.g. ``DataFormats/common/include/CommonDataFormat/EvIndex.h`` and its usage in the ``DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/ROFRecord.h``